### PR TITLE
Implement Mocking the API

### PR DESCRIPTION
**Pull request by bcarterterp**
- MockInterceptor now looks through a list of FileFinders to check for valid URIs to be mocked
- MockInterceptor now integrated with DI
- Updated ReadME for how to mock the API

### CODE REVIEW

File: lib/data/source/network/mock/spoonacular_file_finder.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+/// This [FileFinder] helps in figuring out which file to deliver for spoonacular API calls.
+/// Feel free to configure this however you would like!
+class SpoonacularFileFinder extends FileFinder {
+  final String host = 'api.spoonacular.com';
+  final String path = '/recipes/random';
+
+  @override
+  String? getJsonPath(Uri uri) {
+    if (host != uri.host) return null;
```
Comment (Sagethompson99) [inline lib/data/source/network/mock/spoonacular_file_finder.dart:11]: why not do an `if (host != uri.host || path != uri.path) return null;`

Comment (bcarterterp) [reply lib/data/source/network/mock/spoonacular_file_finder.dart:11]: Agreed, will make that change!

File: lib/main.dart [hunk 2]
```diff
@@ -10,6 +10,8 @@ FutureOr<void> main(List<String> arguments) async {
   const flavor = String.fromEnvironment('FLAVOR', defaultValue: 'dev');
   if (flavor == "prod") {
```
Comment (cxw320) [inline lib/main.dart:11]: Should we try a switch statement here for efficiency? 
```
switch (flavor) {
 case "prod":
   FlavorConfig.flavor = Flavor.prod;
   break;
 case "mock":
   FlavorConfig.flavor = Flavor.mock;
   break;
 default:
   FlavorConfig.flavor = Flavor.dev;
   break;
}
```

File: lib/util/flavor/flavor.dart [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
   dev,
   prod,
+  mock
+}
+
+enum ApiEnv {
+  dev,
+  prod,
+  mock
 }
```
Comment (cxw320) [inline lib/util/flavor/flavor.dart:11]: What do we think about moving this into its own file so devs can find it easier? A bit tougher to look here if they are trying to find the ApiEnv enum specifically.

File: lib/main.dart (hunk 2 above)
Comment (bcarterterp) [reply lib/main.dart]: Done!

**Review by al1baba — APPROVED** [review]

**Review by cxw320 — APPROVED** [review]

**Pull request merged**